### PR TITLE
Adding ferminet model forward pass

### PR DESCRIPTION
**Pull request by shaipranesh2**
## Description
This PR cleaned up the ferminet model forward pass and the gradients!
Fix #2857 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/ferminet.py [hunk 1]
```diff
@@ -26,48 +27,126 @@ def test_f(x: np.ndarray) -> np.ndarray:
 
 class Ferminet(torch.nn.Module):
     """Approximates the log probability of the wave function of a molecule system using DNNs.
-  """
+    """
```
Comment (rbharath) [inline deepchem/models/torch_models/ferminet.py:30]: A quick reminder to add a more fleshed out docstring with a usage example

File: deepchem/models/torch_models/ferminet.py [hunk 2]
```diff
@@ -26,48 +27,126 @@ def test_f(x: np.ndarray) -> np.ndarray:
...
+            for j in range(self.total_electron):
+                self.envelope_w.append(torch.nn.init.kaiming_uniform(torch.empty(n_one[-1],1)).squeeze(-1))
+                self.envelope_g.append(torch.nn.init.uniform(torch.empty(1)).squeeze(0))
+                for k in range(self.nucleon_pos.size()[0]):
+                    self.sigma.append(torch.nn.init.uniform(torch.empty(self.nucleon_pos.size()[0],1)).squeeze(0))
+                    self.pi.append(torch.nn.init.uniform(torch.empty(self.nucleon_pos.size()[0],1)).squeeze(0))
+
+    def forward(self, input):
+        # creating one and two electron features
+        self.input = torch.from_numpy(input)
```
Comment (rbharath) [inline deepchem/models/torch_models/ferminet.py:89]: This is quite complicated. Can we split up into smaller helper functions that have simpler descriptions?
